packages/surveycto/src/Adaptor.js

```diff
@@ -24,92 +21,133 @@ export function execute(...operations) {
   };
 
   return state => {
-    state.configuration.baseUrl = 'https://'.concat(
-      state.configuration.instanceName,
-      '.surveycto.com/api/v1'
-    );
     return commonExecute(...operations)({ ...initialState, ...state });
   };
 }
 
 /**
- * Make a GET request and POST it somewhere else
- * @example
- * execute(
- *   fetch(params)
- * )(state)
+ * Options provided to the HTTP request
+ * @typedef {Object} FormSubmissionOptions
+ * @property {string} [date=0] - Form completion or submission date. Default to `0` which will return all submission data
+ * @property {string} [format='json'] - Form response type, It can be in `csv` or `json`. Default to `json`
+ * @property {string} status - (Opt)Review status. Can be either, `approved`, `rejected`, `pending` or combine eg `approved|rejected`.
+ */
+
+/**
+ * Fetch form submissions
+ * @example <caption>Using state lazy load</caption>
+ * fetchSubmissions($.formId || 'test', { date: '2024-04-18' });
+ * @example <caption> With huma readable date</caption>
+ * fetchSubmissions('test', { date: 'Apr 18, 2024 6:26:21 AM' });
+ * @example <caption> Formatting the results to CSV String</caption>
+ * fetchSubmissions('test', { date: '2024-04-20', format: 'csv' });
+ * @example <caption> With reviewStatus filter</caption>
+ * fetchSubmissions('test', {
+ *   date: '2024-04-18',
+ *   status: 'approved|rejected',
+ * });
+ * @example <caption> With access to the callback</caption>
+ * fetchSubmissions(
+ *   'test',
+ *   {
+ *     date: 'Apr 18, 2024 6:26:21 AM',
+ *     status: 'approved|rejected',
+ *   },
+ *   state => {
+ *     console.log('Hello from the callback!');
+ *     return state;
+ *   }
+ * );
  * @function
- * @param {object} params - data to make the fetch
+ * @param {string} formId - Form id
+ * @param {FormSubmissionOptions} options - Form submission date, timestamp, format, status parameters
+ * @param {function} callback - (Optional) Callback function
  * @returns {Operation}
  */
-export function fetchSubmissions(formId, afterDate, postUrl) {
+export function fetchSubmissions(
+  formId,
+  options = { date: 0, format: 'json' },
```

Comment:
Is date optional, or is it always required?

Reply:
Very good question joe 🤔 
Here is what i found out, They have two apiVersions. `v1` and `v2`. 

**In `v1`. date is required. If it's not specified in the query we get.**
```json
{
  "error": {
    "code": 400,
    "message": "Please provide a date using the 'date' parameter.",
    "responseObject": null
  }
}
```
>PS: 
The reason it's default to `0`, is because that's how you get all form submission data in `v1` 

**But in `v2` it's not required. If you don't specify the date, we will get all data by default**

Reply:
@mtuchi confirming `0` is the default value only and users will still have the option to add their own date filter?

Reply:
@aleksa-krolls we have two default values `date: 0`, which will return all form submission and `format: json` which will return the submissions as JSON object. 

The other format is `csv` which will return the result as CSV string